### ISSUE TITLE
[RFE] add support to view OVN DB 

### ISSUE THREAD

**Issue by cpassare** [gmeghnag/omc#119]
When OVNKubernetes is in use the must-gather collect SB and NB DBs from the OVN leaders. Would be nice to have the possibility to query both databases using omc. The databases are collected in the following files:
- network_logs/leader_nbdb.gz
- network_logs/leader_sbdb.gz

The following are the scripts generally used to read locally the databases:
- https://github.com/openshift/dev-knowledge-base/blob/master/scripts/helpers/run_ovndb_in_container.sh
- https://github.com/openshift/dev-knowledge-base/blob/master/scripts/dev-run-ovndb

**Comment by brunogomes011** [comment]
Hello,

To capture the OVN DB information, we need to connect to one ovsdbserver by network or unix sockets. For this, we often need one container start this port or socket. To implement something similar in the supportshell, we need to run containers per end user which does not appear to be a good idea.

If we will run one container to consume this data, in my opinion, it does not make any sense to have this information in the omc as we have other tools to consult this information.

Other option should implement any tool to consume the json data which seems to be one rework as there are already other available tools to check the ovn dbs.

**Comment by palonsoro** [comment]
I agree with @brunogomes011 . At the end of the day, you need to somehow start a local OVSDB instance for the OVN databases to be consumable. 

In addition, trying to parse the JSON in the OVSDB file may not be worth it. AFAIK, it contains the schema plus all the different transactions that happened in time, which may eventually be compacted (or not). In order to have something near to a parseable JSON, we should be able to compact all those transactions, which can be done by `ovsdb-tool compact` command for standalone database (i.e. for OCP 4.14+ , because in 4.12 OVN databases are clustered). However, this means requiring to have `ovsdb-tool` installed, which is not much better than starting a container.

TBH, I think this is out of the scope of omc. The only way I can see where omc can help is by adding something like an "OVN db extraction" command. OVN databases are kind of "buried" inside the must-gather, having a command to locate those from a node and put them in a more comfortable place might add some value. But that's the only interesting thing I believe OMC can do.

**Comment by brunogomes011** [comment]
@palonsoro 
Actually, as an user I would like to consume the OVN DB using OMC tool. THe RFE request seems to be a good idea. I think it would be great and easier than start a container... Currently, I need to download the DBs to my machine and run the ovs-offline to consume the data.
If it would be possible to consume it from support server, it would be quite faster, but it does not appear to be a good idea in OMC architecture perspective.

After your idea from ovsdb-tool, I noticed that we are able to compact the OVN DB (the one that comes with MG) and it seems to be working successfully. 

We are able to consume the OVN DB using JQ commands:

`
$ ovsdb-tool compact ovnkube-node-xxxx_nbdb ovn-nbdb-test
$ cat ovn-nbdb-test | sed -n '4p' | jq -r .Logical_Router | head -n 20
{
  "xxxxx": {
    "static_routes": [
      "set",
      [
        [
          "uuid",
          "xxx"
        ],
        [
          "uuid",
          "xxx"
        ],
        [
          "uuid",
          "xxx"
`

I think that one option would install the ovsdb-tool (with openvswitch package) in our support server and this option should be available without requiring any container or local downloading. The drawback is that we should lost the ovn-nbctl custom commands.

We just need to find some helpful queries with json to consume the DB. I think it is possible. 

Feel free to add your thoughts.

**Comment by palonsoro** [comment]
IMHO we mustn't create a dependency with `ovsdb-tool` . This tool is not to be used by us in our servers only, but for any body from any company willing to debug a must-gather. Remember that `omc` is built also for Mac OS X and Windows.

If you found a way to achieve the same than `ovsdb-tool compact` but using pure golang code, that might then work. Otherwise, implementing any of the suggestions in this RFE would mean for `omc` to no longer be a multi-platform self-contained binary, which IMHO is a no-go (specially when there are well-established alternatives).

**Comment by brunogomes011** [comment]
@palonsoro 
Yes, I agree with you except we find any LIB or integration using go lang like you mentioned.
The easier way to do it currently is to try to run it from support server but without any relation with OMC. I mentioned it in my former comment.
